apis/v1beta1/vspherefailuredomain_types.go

```diff
@@ -88,6 +94,60 @@ type FailureDomainHosts struct {
 	HostGroupName string `json:"hostGroupName"`
 }
 
+// FailureDomainNetwork defines a network configuration that should be used when consuming
+// this failure domain.
+// @rkatz - To be discussed with team, should we just embed NetworkDeviceSpec?
+type FailureDomainNetwork struct {
+	// NetworkName is the network name for this machine's VM.
+	NetworkName string `json:"name,omitempty"`
+
+	// @rkatz - The reason NetworkDeviceSpec is not being copied here, is because DHCP4 and DHCP6 are not pointers there
+	// This means that, in a case like "I want DHCP4 to be enabled on my template" and "I want DHCP4 to be disabled on this failure domain"
+	// we cannot verify if DHCP4 was unset (nil, we don't care about it on failure domain) vs set to false (I want to FORCE it to be false)
+	// DHCP4 is a flag that indicates whether or not to use DHCP for IPv4
```

Comment:
```suggestion
	// DHCP4 is a flag that indicates whether or not to use DHCP for IPv4
	// @rkatz - The reason NetworkDeviceSpec is not being copied here, is because DHCP4 and DHCP6 are not pointers there
	// This means that, in a case like "I want DHCP4 to be enabled on my template" and "I want DHCP4 to be disabled on this failure domain"
	// we cannot verify if DHCP4 was unset (nil, we don't care about it on failure domain) vs set to false (I want to FORCE it to be false)
```

Needs to get moved around to not break the godoc comment :-) (maybe you wanted to remove the note anyway?)

Reply:
yeap, I wanted to remove :) just left it here to remember myself I wanted to discuss it